treeherder/perf/models.py

```diff
@@ -47,6 +47,8 @@ class PerformanceSignature(models.Model):
     parent_signature = models.ForeignKey('self', on_delete=models.CASCADE, related_name='subtests',
                                          null=True, blank=True)
     has_subtests = models.BooleanField()
+    suite_public_name = models.CharField(max_length=50, blank=True)
+    test_public_name = models.CharField(max_length=50, blank=True)
```

Comment:
In its current form, this PR allows test owners to provide identical public suite & public test names for different signatures from the same framework & repository & platform, which also share the same option collection & the same extra options.

We need to ensure this won't happen, as it will cause confusion & cost us in terms of debug time.
We should provide a test precisely for this aspect & then extend the unique constraints of `PerformanceSignature`, so that this tests passes.

Reply:
Ok.